src/common/test-utils/use-fake-time.ts

```diff
@@ -19,7 +19,13 @@ export const advanceFakeTime = (milliseconds: number) => {
 export const testUsingFakeTime = (dateTime = "2015-10-21T07:28:00Z") => {
   usingFakeTime = true;
 
-  jest.useFakeTimers();
+  jest.useFakeTimers({
+    doNotFake: [
+      "nextTick",
+      "setImmediate",
+      "clearImmediate",
+    ],
+  });
```

Comment:
Hmm. What is the motivation of this? Messing with the default fake time can come as a surprise very easily.

Reply:
So lots of libraries it use those three methods to break causal chains. I think I should be able to revert this if I add the `advance: true` option instead. 

That option automatically advances the fake time 20ms for every real 20ms but does so with an `act()` wrapping.

WDYT?

Reply:
I came up with an even better plan. One that using the real clock to do `act(() => {})` calls.